corporate/views/webhook.py

```diff
@@ -84,6 +86,17 @@ def stripe_webhook(request: HttpRequest) -> HttpResponse:
         event.object_id = payment_intent.id
         event.save()
         handle_payment_intent_succeeded_event(stripe_payment_intent, event)
+    elif stripe_event.type == "invoice.payment_succeeded":
```

Comment:
We should also listen for invoice payment failures or what the event for invoices is that we are not able to collect so that we can delete the `CustomerPlan` corresponding to it.

Reply:
We create `CustomerPlan` only after we receive `invoice.payment_succeeded` event.

We create a `CustomerPlanOffer` entry when we configure fixed-price plan via /support. That was one of the motivation behind behind creating this new model. `CustomerPlan` will be clean.

Reply:
Oh! I see.

Also, we should use `invoice.paid` event here.

Reply:
https://stripe.com/docs/api/events/types#event_types-invoice.paid